pylib/System/TestDef.py

```diff
@@ -582,6 +582,21 @@ def configTest(self):
                 print("ERROR: %s"%l)
             sys.exit(1)
 
+        # find all the sections that match the wild card and expand them
+        # this is simple wild carding, ie *text, text*, *text* and * 
+        # should all work, text*text will not
+        if sections is not None:
+            expsec = []
+            cpsections = list(sections)
+            for sec in cpsections:
+                if '*' in sec:
+                    modsec = sec.replace('*','')
+                    for s in self.config.sections():
+                        if modsec in s:
```

Comment:
a simple replace and grep won't work in all cases.

For example:
> `--skip-sections='*skip'` won't work for sections:
> * TestRun:skip_some_other_section
> * TestRun:please_skip

Maybe instead, do this:

```
if sections is not None:
    expsec = []
    cpsections = list(sections)
    for sec in cpsections:
        if '*' in sec:
            modsec = sec.split('*')
            startswith = modsec[0]
            endswith = modsec[-1]
            findsec = modsec[1:-1]
            for s in allsections:
                if not s.startswith(startswith):
                    continue
                if not s.endswith(endswith):
                    continue
                found = True
                s_tmp = s
                for f in findsec:
                    if not f in s_tmp:
                        found = False
                        break
                    s_tmp = f.join(s_tmp.split(f)[1:])
                if not found:
                    continue
                expsec.append(s)
            sections.remove(sec)
    sections = sections + expsec
```

Reply:
I tested this:

> rtbarell:~> python play.py
> filterSecs("*",['frontback', 'fronthi', 'hiback', 'fronthi2back', 'fronthi2', 'hi2back', 'hi2', 'hi'])
> ['frontback', 'fronthi', 'hiback', 'fronthi2back', 'fronthi2', 'hi2back', 'hi2', 'hi']
> 
> filterSecs("front*",['frontback', 'fronthi', 'hiback', 'fronthi2back', 'fronthi2', 'hi2back', 'hi2', 'hi'])
> ['frontback', 'fronthi', 'fronthi2back', 'fronthi2']
> 
> filterSecs("*back",['frontback', 'fronthi', 'hiback', 'fronthi2back', 'fronthi2', 'hi2back', 'hi2', 'hi'])
> ['frontback', 'hiback', 'fronthi2back', 'hi2back']
> 
> filterSecs("*hi2*",['frontback', 'fronthi', 'hiback', 'fronthi2back', 'fronthi2', 'hi2back', 'hi2', 'hi'])
> ['fronthi2back', 'fronthi2', 'hi2back', 'hi2']
> 
> filterSecs("*hi*back",['frontback', 'fronthi', 'hiback', 'fronthi2back', 'fronthi2', 'hi2back', 'hi2', 'hi'])
> ['hiback', 'fronthi2back', 'hi2back']
>

Reply:
ok, I like this too